File: DeviceManagementDB/Models/Device.cs
using System;
using System.Collections.Generic;

namespace DeviceManagementDB.Models;

public partial class Device
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public int IdManufacturer { get; set; }

    public int IdDeviceType { get; set; }

    public int IdOsversion { get; set; }

    public int IdProcessor { get; set; }

    public int IdRamamount { get; set; }

    public int? IdCurrentUser { get; set; }

    public virtual User? IdCurrentUserNavigation { get; set; }

    public virtual DeviceType IdDeviceTypeNavigation { get; set; } = null!;

    public virtual Manufacturer IdManufacturerNavigation { get; set; } = null!;

    public virtual OperatingSystemVersion IdOsversionNavigation { get; set; } = null!;

    public virtual Processor IdProcessorNavigation { get; set; } = null!;

    public virtual Ramamount IdRamamountNavigation { get; set; } = null!;
}

File: DeviceManagementUI/DeviceManagementUI/src/app/devices/device.ts
import { User } from "../users/user";

export interface Device{
    id: number;
    name: string;
    manufacturer: Manufacturer;
    deviceType: DeviceType;
    osVersion: OperatingSystemVersion;
    processor : Processor;
    ramAmount : RamAmount;
    user: User;  
}

export interface Manufacturer{
    id: number;
    name: string;
}

export interface DeviceType{
    id: number;
    name: string;
}
export interface OperatingSystemVersion{
    id: number;
    name: string;
    os: OperatingSystem;
}

export interface OperatingSystem{
    id: number;
    name: string;
}
export interface Processor{
    id: number;
    name: string;
}
export interface RamAmount{
    id: number;
    amount: number;
}
